pandas/tests/tools/test_to_datetime.py

```diff
@@ -1916,6 +1915,12 @@ def test_to_datetime_with_apply(self, cache):
         result = td.apply(to_datetime, format="%b %y", cache=cache)
         tm.assert_series_equal(result, expected)
 
+    def test_to_datetime_timezone_name(self):
+        # https://github.com/pandas-dev/pandas/issues/49748
+        result = to_datetime("2020-01-01 00:00:00UTC", format="%Y-%m-%d %H:%M:%S%Z")
```

Comment:
Does this only work for UTC or are other names supported e.g. some tz names in https://github.com/pandas-dev/pandas/issues/18702

Reply:
Those would error in both pandas and the stdlib:
```python
In [1]: dt.datetime.strptime('2020-01-01 00:00:00MDT', '%Y-%m-%d %H:%M:%S%Z')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In [1], line 1
----> 1 dt.datetime.strptime('2020-01-01 00:00:00MDT', '%Y-%m-%d %H:%M:%S%Z')

File ~/mambaforge/envs/pandas-dev/lib/python3.8/_strptime.py:568, in _strptime_datetime(cls, data_string, format)
    565 def _strptime_datetime(cls, data_string, format="%a %b %d %H:%M:%S %Y"):
    566     """Return a class cls instance based on the input string and the
    567     format string."""
--> 568     tt, fraction, gmtoff_fraction = _strptime(data_string, format)
    569     tzname, gmtoff = tt[-2:]
    570     args = tt[:6] + (fraction,)

File ~/mambaforge/envs/pandas-dev/lib/python3.8/_strptime.py:349, in _strptime(data_string, format)
    347 found = format_regex.match(data_string)
    348 if not found:
--> 349     raise ValueError("time data %r does not match format %r" %
    350                      (data_string, format))
    351 if len(data_string) != found.end():
    352     raise ValueError("unconverted data remains: %s" %
    353                       data_string[found.end():])

ValueError: time data '2020-01-01 00:00:00MDT' does not match format '%Y-%m-%d %H:%M:%S%Z'

In [2]: to_datetime('2020-01-01 00:00:00MDT', format='%Y-%m-%d %H:%M:%S%Z')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In [2], line 1
----> 1 to_datetime('2020-01-01 00:00:00MDT', format='%Y-%m-%d %H:%M:%S%Z')

File ~/pandas-dev/pandas/core/tools/datetimes.py:1112, in to_datetime(arg, errors, dayfirst, yearfirst, utc, format, exact, unit, infer_datetime_format, origin, cache)
   1110         result = convert_listlike(argc, format)
   1111 else:
-> 1112     result = convert_listlike(np.array([arg]), format)[0]
   1113     if isinstance(arg, bool) and isinstance(result, np.bool_):
   1114         result = bool(result)  # TODO: avoid this kludge.

File ~/pandas-dev/pandas/core/tools/datetimes.py:429, in _convert_listlike_datetimes(arg, format, name, tz, unit, errors, infer_datetime_format, dayfirst, yearfirst, exact)
    426         require_iso8601 = not infer_datetime_format
    428 if format is not None and not require_iso8601:
--> 429     res = _to_datetime_with_format(
    430         arg, orig_arg, name, tz, format, exact, errors, infer_datetime_format
    431     )
    432     if res is not None:
    433         return res

File ~/pandas-dev/pandas/core/tools/datetimes.py:538, in _to_datetime_with_format(arg, orig_arg, name, tz, fmt, exact, errors, infer_datetime_format)
    535         return _box_as_indexlike(result, utc=utc, name=name)
    537 # fallback
--> 538 res = _array_strptime_with_fallback(
    539     arg, name, tz, fmt, exact, errors, infer_datetime_format
    540 )
    541 return res

File ~/pandas-dev/pandas/core/tools/datetimes.py:473, in _array_strptime_with_fallback(arg, name, tz, fmt, exact, errors, infer_datetime_format)
    470 utc = tz == "utc"
    472 try:
--> 473     result, timezones = array_strptime(arg, fmt, exact=exact, errors=errors)
    474 except OutOfBoundsDatetime:
    475     if errors == "raise":

File ~/pandas-dev/pandas/_libs/tslibs/strptime.pyx:145, in pandas._libs.tslibs.strptime.array_strptime()
    143         iresult[i] = NPY_NAT
    144         continue
--> 145     raise ValueError(f"time data '{val}' does not match "
    146                      f"format '{fmt}' (match)")
    147 if len(val) != found.end():

ValueError: time data '2020-01-01 00:00:00MDT' does not match format '%Y-%m-%d %H:%M:%S%Z' (match)
```